### PR TITLE
:sparkles:(metrics): add adaptive calibration plots

### LINKED ISSUE #214: :sparkles: Add Adaptive Calibration plots
For now, `.plot()` returns `NotImplementedError`. We should implement a plotting system for the Adaptive Calibration Error too.

### PR DESCRIPTION
**Pull request by o-laurent**

### CODE REVIEW

File: src/torch_uncertainty/metrics/classification/calibration/adaptive_calibration_error.py [hunk 1]
```diff
@@ -45,6 +49,247 @@ def _equal_binning_bucketize(
...
+
+    ax.text(
+        0.98,
+        0.02,
+        f"ACE={ace:.02%}",
+        color="black",
+        ha="right",
+        va="bottom",
+        transform=ax.transAxes,
+    )
```
[the remark this answers is not in the document]
Comment (o-laurent) [reply src/torch_uncertainty/metrics/classification/calibration/adaptive_calibration_error.py]: @alafage this comment is likely legitimate. Shouldn't we pass the computed value and the metric dimension (l1, l2, linf) to the plotting function to ensure consistency? This would mean refactoring somewhat, and that's something I can do in this PR too.

**Comment by o-laurent** [comment]
Another thing that we should maybe try to work on is reducing the code duplication, which is pretty high here.

File: src/torch_uncertainty/metrics/classification/calibration/adaptive_calibration_error.py [hunk 2]
```diff
@@ -45,6 +49,247 @@ def _equal_binning_bucketize(
...
+    bin_lowers = torch.stack([cb[0] for cb in conf_splits])
+    bin_uppers = torch.stack([cb[-1] for cb in conf_splits])
+
+    return (
+        pad_sequence(acc_splits, batch_first=True).sum(1) / count_bin,
+        pad_sequence(conf_splits, batch_first=True).sum(1) / count_bin,
+        count_bin / len(confidences),
+        bin_lowers,
+        bin_uppers,
+    )
```
[the remark this answers is not in the document]
Comment (o-laurent) [reply src/torch_uncertainty/metrics/classification/calibration/adaptive_calibration_error.py:86]: We should probably use `_equal_binning_bucketize_with_bounds` only @alafage, creating issue #300

**Review by alafage — APPROVED** [review]
